modules/widgets/top-posts.php

```diff
@@ -143,12 +187,31 @@ function widget( $args, $instance ) {
 			} else {
 				$get_image_options['avatar_size'] = 40;
 			}
+			/**
+			 * Top Posts Widget Image options.
+			 *
+			 * @since 1.8.0
+			 *
+			 * @param array $get_image_options {
+			 * Array of Image options.
+			 * @type bool true Should we default to Gravatars when no image is found? Default is true.
+			 * @type string $gravatar_default Default Image URL if no Gravatar is found.
+			 * @type int $avatar_size Default Image size.
+			 * }
+			 */
 			$get_image_options = apply_filters( 'jetpack_top_posts_widget_image_options', $get_image_options );
 		}
 
 		$posts = $this->get_by_views( $count );
 
-		if ( !$posts ) {
+		// Filter the returned posts. Remove all posts that do not match the chosen Post Types.
+		foreach ( $posts as $k => $post ) {
+			if ( isset( $types ) && ! in_array( $post['post_type'], $types ) ) {
```

Comment:
If `$types` is not set, then this whole cycle is pointless. The check should be done outside the cycle, don't you think?